pytools/persistent_dict.py

```diff
@@ -449,38 +343,18 @@ class NoSuchEntryError(KeyError):
     pass
 
 
-class NoSuchEntryInvalidKeyError(NoSuchEntryError):
-    """Raised when an entry is not found in a :class:`PersistentDict` due to an
-    invalid key file."""
-    pass
-
-
-class NoSuchEntryInvalidContentsError(NoSuchEntryError):
-    """Raised when an entry is not found in a :class:`PersistentDict` due to an
-    invalid contents file."""
-    pass
-
-
-class NoSuchEntryCollisionError(NoSuchEntryError):
-    """Raised when an entry is not found in a :class:`PersistentDict`, but it
-    contains an entry with the same hash key (hash collision)."""
-    pass
-
```

Comment:
Existing code may import these, so we're not really at liberty to remove them. Why are you proposing to remove them?

Reply:
`NoSuchEntryInvalidKeyError` and `NoSuchEntryInvalidContentsError` can't occur anymore, because we don't have separate key and content files with this PR. We could provide a custom exception for something like 'invalid sqlite file', but I'm not sure that would be more helpful for the application.

Regarding `NoSuchEntryCollisionError`, we currently have no way of checking for collisions, since the key itself is not stored in the sqlite file, just its hash.

Reply:
https://github.com/inducer/pytools/commit/367683c6eaac0b0cc2a6048d3de5b30d45a0605a & https://github.com/inducer/pytools/commit/7028e6bbd63ef734e5d23cb8b144f06dcb2153c2 restore the collision tests.